### ISSUE TITLE
Ensure Builder API can instantiate custom packages paths

### ISSUE THREAD

**Issue by guybedford** [jspm/jspm-cli#931]
```javascript
new require('jspm').Builder({
  paths: {
    'github:*': '../../path/to/github/*'
  }
});
```

Should be able to define server-specific paths for builds. This ability should also be carefully documented as the primary mechanism. We should potentially throw on further configs of existing paths as well when past the point of altering them.

**Comment by lookfirst** [comment]
@guybedford  

With the `lookfirst/broken` directory structure and configuration, if `src/main/jspm_packages` is missing then I get this error... notice how it is trying to create a directory in `target/jspm_packages`?! That exists cause it is a symlink to `src/main/jspm_packages`...

```
jon:hub jon$ rm -rf src/main/jspm_packages
jon:hub jon$ jspm install
     Looking up npm:babel-core
     Looking up npm:babel-runtime
     Looking up npm:clean-css
     Looking up npm:core-js
     Looking up npm:lodash
     Updating registry cache...
     Looking up github:angular/bower-angular
     Looking up github:eight04/angular-datetime
     Looking up github:revolunet/angular-carousel
     ...

err  Error: EEXIST, mkdir '/Users/jon/checkout/hub/target/jspm_packages'
         at Error (native)
```

**Comment by lookfirst** [comment]
If I do, `mkdir src/main/jspm_packages`, then the above problem goes away.

**Comment by lookfirst** [comment]
I wrote some comments... figured some stuff out... ignore them. =)

**Comment by guybedford** [comment]
Released in jspm@0.16.0-beta.6.
